### PR TITLE
Packet

### PR DESCRIPTION
**Pull request by evelynliu16**
Finished packet.py file which decodes the message into a packet object.

### CODE REVIEW

File: packet.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        old_sum = data[tracker + 1:len(data) - 1]
+
+        # Get sensor id.
+        i = 0
+        while data[i] != ",":
+            i += 1
+        sensor_id = int(data[1:i])
+
+        # Calculate new checksum and compare.
+        checksum_data = data[i + 1:tracker]
```
Comment (fazerlicourice7) [inline packet.py]: You should go from `1: tracker` since the checksum includes the sensor id

File: packet.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        new_sum = fletcher16(checksum_data)
+        if old_sum == new_sum:
+            print("Check sum succeeded")
+        else:
+            print("New sum does not match the old sum")
+
+        # Get sensor data.
+        data = checksum_data.split(",")
+        data = [int(data[i]) for i in range(len(data))]
```
Comment (fazerlicourice7) [inline packet.py]: it could be floats.

**Review by fazerlicourice7 — CHANGES_REQUESTED** [review]
Very small changes, but overall looks great!

**Comment by vamshiBalanaga** [comment]
Can we also add an `encode_packet` func that takes in the sensor id and an array of floats and calculates the checksum and returns a string packet

**Review by fazerlicourice7 — CHANGES_REQUESTED** [review]
Can the encode just be added to the same Packet class? I don't see why there should be two of them. just allow us to create it with either the packet string or the raw initial data.

File: packet.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        self.sensor_id, self.data, self.checksum = self.decode_message(data)
+
+    def __init__(self, id, data):
+        self.sensor_id = id
+        self.data = data
+        encoding = str(id)
+        for i in data:
+            encoding += "," + str(i)
+        checksum = fletcher16(encoding)
+        self.checksum = checksum
```
Comment (fazerlicourice7) [inline packet.py]: Move these into `encode_data` and here, just call encode data with the right parameters.
Add a string variable that holds the encoded message so after I create a packet instance with 
```
packet = Packet(3, {23, 43, 5})
packet.encoded_message
```

**Review by fazerlicourice7 — CHANGES_REQUESTED** [review]

**Review by fazerlicourice7 — APPROVED** [review]

**Pull request merged**